core/primitives/src/block.rs

```diff
@@ -596,6 +607,13 @@ impl Block {
             .0
     }
 
+    pub fn compute_challenges_root(challenges: &Challenges) -> CryptoHash {
+        let mut arr =
+            challenges.iter().map(|challenges| challenges.hash).collect::<Vec<CryptoHash>>();
+        arr.sort();
```

Comment:
What does sort do here?

Reply:
> What does sort do here?

I've added `test_elements_order` to merkle to show you. Should block hash be depended on order of challenges?

Reply:
I think it should because a block is produced by a given block producer and what we want to make sure is that other people cannot also produce blocks that have the same hash.

Reply:
Okay, we can leave them unsorted.
Is it okay that having Block header, chunks and challenges it may be impossible to get same block hash because of order of challenges?